bundler/lib/bundler/compact_index_client/cache.rb

```diff
@@ -7,114 +7,107 @@ class CompactIndexClient
     class Cache
       attr_reader :directory
 
-      def initialize(directory)
+      def initialize(directory, fetcher = nil)
         @directory = Pathname.new(directory).expand_path
-        info_roots.each {|dir| mkdir(dir) }
-        mkdir(info_etag_root)
+        @updater = Updater.new(fetcher) if fetcher
+        @mutex = Thread::Mutex.new
+        @endpoints = Set.new
+
+        SharedHelpers.filesystem_access(@directory) do
```

Comment:
Each `mkdir` already wraps the directory creation with this method, is this one necessary?

Reply:
hmm, good point. I pushed this down from Fetcher::CompactIndex. I think maybe it's just entirely unnecessary.